src/map/skill.c

```diff
@@ -1084,73 +1085,35 @@ static int skill_get_range2(struct block_list *bl, int skill_id, int skill_lv)
 		range *=-1;
 	}
 
-	//TODO: Find a way better than hardcoding the list of skills affected by AC_VULTURE
-	switch( skill_id ) {
-		case AC_SHOWER:
-		case MA_SHOWER:
-		case AC_DOUBLE:
-		case MA_DOUBLE:
-		case HT_BLITZBEAT:
-		case AC_CHARGEARROW:
-		case MA_CHARGEARROW:
-		case SN_FALCONASSAULT:
-		case HT_POWER:
-		/**
-		 * Ranger
-		 **/
-		case RA_ARROWSTORM:
-		case RA_AIMEDBOLT:
-		case RA_WUGBITE:
-			if (sd != NULL)
-				range += pc->checkskill(sd, AC_VULTURE);
-			else
-				range += battle->bc->mob_eye_range_bonus;
-			break;
-		// added to allow GS skills to be effected by the range of Snake Eyes [Reddozen]
-		case GS_RAPIDSHOWER:
-		case GS_PIERCINGSHOT:
-		case GS_FULLBUSTER:
-		case GS_SPREADATTACK:
-		case GS_GROUNDDRIFT:
-			if (sd != NULL)
-				range += pc->checkskill(sd, GS_SNAKEEYE);
-			else
-				range += battle->bc->mob_eye_range_bonus;
-			break;
-		case NJ_KIRIKAGE:
-			if (sd != NULL)
-				range = skill->get_range(NJ_SHADOWJUMP, pc->checkskill(sd, NJ_SHADOWJUMP));
-			break;
-		/**
-		 * Warlock
-		 **/
-		case WL_WHITEIMPRISON:
-		case WL_SOULEXPANSION:
-		case WL_MARSHOFABYSS:
-		case WL_SIENNAEXECRATE:
-		case WL_DRAINLIFE:
-		case WL_CRIMSONROCK:
-		case WL_HELLINFERNO:
-		case WL_COMET:
-		case WL_CHAINLIGHTNING:
-		case WL_TETRAVORTEX:
-		case WL_EARTHSTRAIN:
-		case WL_RELEASE:
-			if (sd != NULL)
-				range += pc->checkskill(sd, WL_RADIUS);
-			break;
-		/**
-		 * Ranger Bonus
-		 **/
-		case HT_LANDMINE:
-		case HT_FREEZINGTRAP:
-		case HT_BLASTMINE:
-		case HT_CLAYMORETRAP:
-		case RA_CLUSTERBOMB:
-		case RA_FIRINGTRAP:
-		case RA_ICEBOUNDTRAP:
-			if (sd != NULL)
-				range += (1 + pc->checkskill(sd, RA_RESEARCHTRAP))/2;
+	inf = skill->get_inf2(skill_id);
+
+	if (inf & INF2_RANGE_VULTURE) {
+		if (sd != NULL)
```

Comment:
for avoid multiply checks for ``sd != NULL`` probably better move sd check before ``inf`` checks.
should be something like this
```
if (sd != NULL) {
// inf checks where sd is not NULL
} else {
// inf checks where sd is NULL
}
```

Reply:
great idea <3